datapackage_pipelines_mojp/common/processors/geocode.py

```diff
@@ -0,0 +1,123 @@
+from datapackage_pipelines.wrapper import ingest, spew
+from datapackage_pipelines_mojp.common.constants import GEOCODE_LOCATION_SCHEMA
+import requests, logging, datetime, os
+import geocoder
+from sqlalchemy import create_engine, MetaData, or_
+from sqlalchemy.orm import sessionmaker
+
+
+# attributes in entity details that might contain the full address in a single string
+LOCATION_ADDRESS_FIELDS = ["location"]
+
+# simple mapping of fields to their possible sources in the data
+# we will take the first item from the list of possible sources that has some data
+LOCATION_FIELDS_MAP = {"street": ["street"],
+                       "house_number": ["street_number", "house_number"],
+                       "city": ["city"],
+                       "country": ["country"],
+                       "po_box": ["pob", "zipcode", "postal_code", "pob_postal_code"],}
+
+
+class GeoCodeEntities(object):
+
+    def __init__(self, parameters, datapackage, resources):
+        self.parameters = parameters
+        self.datapackage = datapackage
+        self.resources = resources
+        self.locations_cache = {}
+
+    @staticmethod
+    def get_schema():
+        return GEOCODE_LOCATION_SCHEMA
+
+    def initialize_db_session(self):
+        connection_string = os.environ.get("DPP_DB_ENGINE")
+        assert connection_string is not None, \
+            "Couldn't connect to DB - " \
+            "Please set your '%s' environment variable" % "DPP_DB_ENGINE"
+        engine = create_engine(connection_string)
+        return sessionmaker(bind=engine)()
+
+    # returns a location string to geocode by
+    # returns None if no suitable location string is found
+    def get_location_string(self, entity_details):
```

Comment:
this function is not needed, we get the location string as-is from the source data